src/ModelContextProtocol/Client/McpClientExtensions.cs

```diff
@@ -52,7 +52,7 @@ public static async Task<IList<McpClientTool>> ListToolsAsync(
         {
             var toolResults = await client.SendRequestAsync(
                 RequestMethods.ToolsList, 
-                CreateCursorDictionary(cursor),
+                CreateCursorDictionary(cursor)!,
```

Comment:
Why is the null-forgiving operator now necessary here?

Reply:
I'm not sure why it's only showing up now (maybe with the net9.0 addition), but the source generated type info properties are shared regardless of the nullability applied to their members, and in here that nullability differs from the other arguments.

Reply:
Ah right, it's because pre-9.0 generated code doesn't use nullable-oblivious annotations for its JsonTypeInfos.